cmd/resolve_discussion.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type DiscussionResolveRequest struct {
+	DiscussionID string `json:"discussion_id"`
+	Resolved     bool   `json:"resolved"`
+}
+
+func DiscussionResolveHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPut:
+		DiscussionResolve(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+func DiscussionResolve(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	c := r.Context().Value("client").(Client)
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		c.handleError(w, err, "Could not read request body", http.StatusBadRequest)
+		return
+	}
+	defer r.Body.Close()
+	var resolveDiscussionRequest DiscussionResolveRequest
+	err = json.Unmarshal(body, &resolveDiscussionRequest)
+	if err != nil {
+		c.handleError(w, err, "Could not read JSON from request", http.StatusBadRequest)
+		return
+	}
+
+	_, res, err := c.git.Discussions.ResolveMergeRequestDiscussion(
+		c.projectId,
+		c.mergeId,
+		resolveDiscussionRequest.DiscussionID,
+		&gitlab.ResolveMergeRequestDiscussionOptions{Resolved: &resolveDiscussionRequest.Resolved},
+	)
+
+	if err != nil {
+		c.handleError(w, err, "Could not update resolve status of discussion", res.StatusCode)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	response := SuccessResponse{
+		Message: "Resolve status of discussion updated",
```

Comment:
Can we be specific with this message? We have the boolean so we could easily write "Discussion resolved" or "Discussion unresolved" here.

Reply:
sure :+1: